modular_skyrat/modules/milker/code/milker.dm

```diff
@@ -0,0 +1,305 @@
+#define VOLUME_LIGHT 1 // Volume 'light' mode extracts per action
+#define VOLUME_STRONG 5 // Volume 'strong' mode extracts per action
+#define HANDLE_LIGHT 1 // 'light' mode
+#define HANDLE_STRONG 2 // 'strong' mode
+#define HANDLE_MODE_LOWEST 1 // The lowest numbered setting currently available (so we dont need yet another list)
+#define HANDLE_MODE_HIGHEST 2 // The highest numbered setting currently available
+#define PLEASURE_LIGHT 1 // How much pleasure is gained through 'light' mode
+#define PLEASURE_STRONG 3 // How much pleasure is gained through 'strong' mode
+#define AROUSAL_LIGHT 1 // How much arousal is gained through 'light' mode
+#define AROUSAL_STRONG 3 // How much arousal is gained through 'strong' mode
+#define PAIN_LIGHT 3 // How much pain is gained through 'light' mode
+#define PAIN_STRONG 10 // How much pain is gained through 'strong' mode
+// Self-(and otherwise) Expression (of mammary organs)
+
+/// The expression device
+/obj/item/milker
+	name = "milker"
+	desc = "You and me, baby, ain't nothin' but mammals."
+	icon_state = "latexballon"
+	inhand_icon_state = "nothing"
+	force = 0
+	throwforce = 0
+	item_flags = DROPDEL | ABSTRACT | HAND_ITEM
+	attack_verb_continuous = list("slaps")
+	attack_verb_simple = list("slap")
+	hitsound = 'sound/effects/snap.ogg'
+	var/pleasure_amt = PLEASURE_LIGHT
+	var/arousal_amt = AROUSAL_LIGHT
+	var/pain_amt = PAIN_LIGHT // *chomp*
+	var/handle_mode = HANDLE_LIGHT // How aggressively we're going to milk someone
+	var/squirt_volume = VOLUME_LIGHT // How much we're trying to express per squeeze / suck / CHOMP
+
+/obj/item/milker/Initialize(mapload)
+	. = ..()
+	toggleMode(null, TRUE)
+
+/obj/item/milker/attack(mob/living/M, mob/living/carbon/human/user)
+	if(!in_range(M, user))
+		to_chat(user, span_warning("[M] is too far away!"))
+		return FALSE
+	var/mob/living/carbon/being_milked = M
+
+	var/obj/item/organ/genital/breasts/breasts = being_milked.getorganslot(ORGAN_SLOT_BREASTS)
+	if(!breastCheck(being_milked, user, breasts))
+		return FALSE
+
+	/// What's in our other hand? If it's a reagent container, we could try bottling it!
+	/// Holding a reagent container implies wanting to *use* the container, so if the container doesn't work, don't proceed!
+	var/obj/item/reagent_containers/milk_bottle = user.get_inactive_held_item()
+	if(istype(milk_bottle))
+		bottleMilk(being_milked, user, breasts, milk_bottle)
+	else // Or just drink it
+		drinkMilk(being_milked, user, breasts)
+
+/// Set how vigorously you want to make the stuff come out
+/obj/item/milker/attack_self(mob/user)
+	toggleMode(user, FALSE)
+
+/obj/item/milker/proc/toggleMode(mob/user, milker_setup)
```

Comment:
snake_case, not lowerCamelCase, please :)
Also, add documentation to new procs you create if you can :)

Reply:
Done, forgot what the standard was.